controllers/upgrade_handlers.go

```diff
@@ -76,5 +172,146 @@ func (r *ImageBasedUpgradeReconciler) handleUpgrade(ctx context.Context, ibu *lc
 		metav1.ConditionFalse,
 		"Upgrade completed",
 		ibu.Generation)
+
 	return doNotRequeue(), nil
 }
+
+// TriggerBackup apply backup CRs if possible to start backup
+func (r *ImageBasedUpgradeReconciler) TriggerBackup(ctx context.Context, backupGroups [][]*velerov1.Backup) error {
+	for _, backupGroup := range backupGroups {
+		for _, backup := range backupGroup {
+			var existingBackup *velerov1.Backup
+			existingBackup, err := backuprestore.GetBackup(ctx, r.Client, backup.Name, backup.Namespace)
+			if err != nil {
+				return err
+			}
+
+			if existingBackup == nil {
+				// Backup has not been applied. Applying backup
+				err := r.CreateNewBackupCr(ctx, backup)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+func (r *ImageBasedUpgradeReconciler) CreateNewBackupCr(ctx context.Context, backup *velerov1.Backup) error {
```

Comment:
As we talked, this shouldn't be splitted out of the triggerBackup function. If the backup requires ordering, the previous group should be completed prior to applying the second group.

Reply:
In addition to that, if the upgrade handler is an one-shot, you will need to periodically check for the resulf of the backup CR.  If it's not an one-shot, you will return requeue interval and check the result in the next intervel.

Reply:
Reverted + updated as discussed